ddtrace/appsec/_iast/_taint_tracking/Aspects/AspectSlice.cpp

```diff
@@ -0,0 +1,96 @@
+#include "AspectSlice.h"
+
+TaintRangeRefs
+reduce_ranges_from_index_range_map(TaintRangeRefs index_range_map)
+{
+    TaintRangeRefs new_ranges;
+    TaintRangePtr current_range;
+    size_t current_start = 0;
+    size_t index;
+
+    for (index = 0; index < index_range_map.size(); ++index) {
+        auto taint_range{ index_range_map.at(index) };
+        if (taint_range != current_range) {
+            if (current_range) {
+                new_ranges.emplace_back(initializer->allocate_taint_range(current_start, index, current_range->source));
+            }
+            current_range = taint_range;
+            current_start = index;
+        }
+    }
+    if (current_range != nullptr) {
+        new_ranges.emplace_back(initializer->allocate_taint_range(current_start, index, current_range->source));
+    }
+    return new_ranges;
+}
+
+TaintRangeRefs
+build_index_range_map(PyObject* text, TaintRangeRefs& ranges, PyObject* start, PyObject* stop, PyObject* step)
+{
+    TaintRangeRefs index_range_map;
+    long long index = 0;
+    for (const auto& taint_range : ranges) {
+        auto shared_range = taint_range;
+        while (index < taint_range->start) {
+            index_range_map.emplace_back(nullptr);
+            index++;
+        }
+        while (index < (taint_range->start + taint_range->length)) {
+            index_range_map.emplace_back(shared_range);
+            index++;
+        }
+    }
+    while (index < (long long)py::len(text)) {
```

Comment:
```suggestion
    while (index < PyUnicode_GET_LENGTH(text)) {
```
You can use `PyUnicode_GET_LENGTH`, should be faster than `pybind11`, it returns `size_t`, not `long long`

Reply:
Oh, but you would need to check if it's bytes or unicode and use `PyBytes_Size` for bytes